infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/IngressServerExposer.java

```diff
@@ -39,16 +42,22 @@
 
   private final ExternalServiceExposureStrategy serviceExposureStrategy;
   private final Map<String, String> ingressAnnotations;
+  private final Map<String, String> labels;
   private final String pathTransformFmt;
 
   @Inject
   public IngressServerExposer(
       ExternalServiceExposureStrategy serviceExposureStrategy,
       @Named("infra.kubernetes.ingress.annotations") Map<String, String> annotations,
+      @Nullable @Named("che.infra.kubernetes.ingress.labels") String labelsProperty,
       @Nullable @Named("che.infra.kubernetes.ingress.path_transform") String pathTransformFmt) {
     this.serviceExposureStrategy = serviceExposureStrategy;
     this.ingressAnnotations = annotations;
     this.pathTransformFmt = pathTransformFmt == null ? PATH_TRANSFORM_PATH_CATCH : pathTransformFmt;
+    this.labels =
+        labelsProperty != null
+            ? Splitter.on(",").withKeyValueSeparator("=").split(labelsProperty)
```

Comment:
IMHO, we should ignore whitespace around commas and at the start and end of the string...